Content.Server/Thief/Systems/ThiefUndeterminedBackpackSystem.cs

```diff
@@ -0,0 +1,85 @@
+using Content.Server.Antag;
+using Content.Server.Thief.Components;
+using Content.Shared.Item;
+using Content.Shared.Thief;
+using Robust.Server.GameObjects;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Thief.Systems;
+public sealed class ThiefUndeterminedBackpackSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly UserInterfaceSystem _ui = default!;
+
+    private const int MaxSelectedSets = 2;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ThiefUndeterminedBackpackComponent, BoundUIOpenedEvent>(OnUIOpened);
+        SubscribeLocalEvent<ThiefUndeterminedBackpackComponent, ThiefBackpackApproveMessage>(OnApprove);
+        SubscribeLocalEvent<ThiefUndeterminedBackpackComponent, ThiefBackpackChangeSetMessage>(OnChangeSet);
+    }
+
+    private void OnUIOpened(Entity<ThiefUndeterminedBackpackComponent> backpack, ref BoundUIOpenedEvent args)
+    {
+        UpdateUI(backpack.Owner, backpack.Comp);
+    }
+
+    private void OnApprove(Entity<ThiefUndeterminedBackpackComponent> backpack, ref ThiefBackpackApproveMessage args)
+    {
+        if (backpack.Comp.SelectedSets.Count != MaxSelectedSets)
+            return;
+        if (!TryComp<UserInterfaceComponent>(backpack.Owner, out var ui))
+            return;
+
+        foreach (var i in backpack.Comp.SelectedSets)
+        {
+            var set = _proto.Index(backpack.Comp.PossibleSets[i]);
+            foreach (var item in set.Content)
+            {
+                var ent = Spawn(item);
+                if (TryComp<ItemComponent>(ent, out var itemComponent))
+                    _transform.DropNextTo(ent, backpack.Owner);
+                else
+                    _transform.SetCoordinates(ent, Transform(backpack.Owner).Coordinates);
+                //SpawnInContainerOrDrop(item, backpack,) TO DO change spawning after delete
```

Comment:
Isn't there a helper method on engine for this now?

Reply:
I can't solve it yet: in one of the kits there is a box that shouldn't appear in the inventory. So far I can't get it to spawn in the location of the backpack, or in the location of the player with the backpack. I'd love a hint!

Reply:
ContainerSystem.CanInsert ?
Check if it can fit in bag or hands, or drop on the floor

Reply:
The problem is not checking the ability to insert.  the problem is to spawn the crate in the required coordinates